### ISSUE TITLE
SPFarmSolution deploys on all web applications

### ISSUE THREAD

**Issue by shurick81** [PowerShell/SharePointDsc#1030]
#### Details of the scenario you tried and the problem that is occurring
We are running a resource for deploying a wsp package to only one SP web application. However, it deploys the WSP to all the applications in the farm.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```PowerShell
                SPFarmSolution contoso.sp2013.kdb
                {
                    Name                    = "contoso.sp2013.kdb.wsp"
                    LiteralPath             = "\\SPADM01\Files\sp2013common\exportedwsp\contoso.sp2013.kdb.wsp"
                    WebAppUrls              = @( "https://$siteCollectionHostNamekdb" )
                    PsDscRunAsCredential    = $SPInstallAccountCredential
                }
```

#### The operating system the target node is running
Windows 2012 R2
#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint Server 2013
#### Version of the DSC module that was used ('dev' if using current dev branch)
3.1.0.0

**Comment by shurick81** [comment]
Ok, there seems to be something relevant in upcoming 3.2.0.0 version, according to change log. However, I could not find any details what stays behind "Fixed issue where uninstalling a solution would not work as expected if it contained web application resources". I could not find anything relevant in GitHub issues.
Any plan when approximately 3.2.0.0 is going to be available via gallery?

**Comment by ykuijs** [comment]
Should be available now

**Comment by ykuijs** [comment]
The update in v3.2 isn't that big. Was just to fix an issue because we implemented Strict-Mode v1 for all Pester tests.

Did you already do some troubleshooting yourself/review the code where it goes wrong?

**Comment by ykuijs** [comment]
@shurick81 Did you already do some troubleshooting yourself?

**Comment by shurick81** [comment]
not really @ykuijs. I tried with 3.2 and the behavior seems the same. We want to deploy the wsp on only one web application, but after running the config, I can see in CA that the solution is deployed to each web application.

**Comment by andikrueger** [comment]
Just out of curiosity: Is the solution configured to be deploy-able to only one web-application? I ran in a similar issue before and the package xml of the wsp was missing a SafeControl entry. See e.g.: http://sharedpointers.blogspot.com/2011/03/deploying-solutions-to-specific-web.html

**Comment by ykuijs** [comment]
@shurick81 @andikrueger is right. Some solutions have to be deployed globally. Is your solution able to be deployed to one web application?

**Comment by ykuijs** [comment]
Would it be possible to share your WSP so I can test on my environment?

**Comment by ykuijs** [comment]
Hi @shurick81, any update on this issue?

**Comment by shurick81** [comment]
@markkusoft, have you got any spare file to share?

**Comment by markkusoft** [comment]
Unfortunately I don't have anything cleared for sharing at this time

**Comment by ykuijs** [comment]
Is it an option to share the manifest.xml of the solution? This can be done via email so this isn't shared publically. I have send @shurick81 an email on April 9th from which you can retrieve my email address :-)

**Comment by ykuijs** [comment]
This issue should be fixed by PR #1108, which is just merged to Dev.

**Comment by shurick81** [comment]
I think it was also related to this behavior: #1221
